### PR TITLE
libmediainfo: add 21.03 + honor debug lib name + bump dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@madebr I've removed 2 patches, I don't see any reason to maintain them forever:
- for pdb files, we can remove them after installation
- I prefer to honor upstream lib names for debug builds

File: recipes/libmediainfo/all/conandata.yml [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
+  "21.03":
+    url: "https://mediaarea.net/download/source/libmediainfo/21.03/libmediainfo_21.03.tar.xz"
+    sha256: "56b7e9abf80cba48032165cd7a46fd8d43dd63e3af35765f66c3f134caaca4ca"
   "20.03":
     url: "https://mediaarea.net/download/source/libmediainfo/20.03/libmediainfo_20.03.tar.xz"
     sha256: "6848310eb7050efe7f90f18191dacfcc220c924390b8088fdf0d553274917f72"
 patches:
   "20.03":
     - patch_file: "patches/0001-cmakelists-CMAKE_MODULE_PATH-find_package.patch"
       base_path: "source_subfolder"
```
Comment (madebr) [inline recipes/libmediainfo/all/conandata.yml:11]: Doesn't 21.03 need this patch?

**Review by madebr — APPROVED** [review]
I like less patches!
Great job @SpaceIm!

File: recipes/libmediainfo/all/conandata.yml (hunk 1 above)
Comment (SpaceIm) [reply recipes/libmediainfo/all/conandata.yml:11]: No, it's fixed

**Comment by SpaceIm** [comment]
Actually, there is an issue with shared debug, they have hardcoded shared name without debug suffix, it's not consistent with their CMakeLists.

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**